core/testcontainers/core/container.py

```diff
@@ -78,6 +83,33 @@ def with_kwargs(self, **kwargs) -> Self:
         self._kwargs = kwargs
         return self
 
+    def with_copy_file_to_container(self, transferable: Transferable) -> Self:
+        self._files.append(transferable)
+
+        return self
+
+    def copy_file_from_container(self, container_file: os.PathLike, destination_file: os.PathLike) -> os.PathLike:
+        tar_stream, _ = self._container.get_archive(container_file)
+
+        for chunk in tar_stream:
+            with tarfile.open(fileobj=io.BytesIO(chunk)) as tar:
+                for member in tar.getmembers():
+                    with open(destination_file, "wb") as f:
+                        f.write(tar.extractfile(member).read())
+
+        return destination_file
+
+    @staticmethod
+    def _put_data_in_container(container, transferable: Transferable):
+        data = io.BytesIO()
+
+        with transferable as f, tarfile.open(fileobj=data, mode="w") as tar:
+            tar.add(f.input_path, arcname=f.output_path)
```

Comment:
IIUC, the temporary file could be avoided if using `tar.addfile` - it is somewhat more tedious to set up a `TarInfo` but maybe worth it to skip the temp file

Reply:
Also, I think all the copy_file APIs in testcontainers accept a file mode. With the current pattern, it can be something like `filter=lambda tarinfo: tarinfo.mode = foo`, but if switching to `addfile` it could be set directly to the `TarInfo`